### PR TITLE
Removing NetCore dupe project and fixing missing task issue.

### PR DESCRIPTION
**Pull request by joperezr**
These changes fix the issues caused by pull request #311 
It includes:
* Upgrading buildtools to version 121
* Removing Microsoft.DotNet.Build.Tasks.NetCore project and keeping just one portable version of it in the solution.
* Removing .depproj, and using Property instead that will deploy the dependencies.
* Adding back Task ZipFileCreateFromDirectory to the task library

### CODE REVIEW

**Comment by joperezr** [comment]
cc @ericstj @weshaggard

**Comment by joperezr** [comment]
@dotnet-bot test this please

**Comment by joperezr** [comment]
@dotnet-bot test this please

File: src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj [hunk 1]
```diff
@@ -23,7 +18,8 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
   </PropertyGroup>
   <ItemGroup>
-    <Compile Include="Delegates.cs" />
+    <Compile Include="Delegates.cs">
+    </Compile>
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj]: Nit: unnecessary change.

File: src/Microsoft.DotNet.Build.Tasks/project.json [hunk 2]
```diff
@@ -1,8 +1,27 @@
...
+    "Microsoft.CSharp": "4.0.0",
+    "System.Linq": "4.0.0",
+    "System.Xml.XDocument": "4.0.0",
+    "System.IO.Compression": "4.0.0",
+    "System.IO.Compression.ZipFile": "4.0.0",
+    "System.Xml.ReaderWriter": "4.0.10",
+    "System.Xml.XPath.XmlDocument": "4.0.0",
+    "System.Reflection": "4.0.10",
+    "System.Reflection.Metadata": "1.1.0-alpha-*",
+    "Newtonsoft.Json": "7.0.1"
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks/project.json]: Previously the packages.config had a dependency on Xunit.Runners.  Was that just cruft?

File: src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj [hunk 3]
```diff
@@ -51,51 +47,7 @@
...
-    <Reference Include="System.IO.Compression" />
-    <Reference Include="System.Reflection.Metadata">
-      <HintPath>..\..\packages\System.Reflection.Metadata.1.0.17-beta\lib\portable-net45+win8\System.Reflection.Metadata.dll</HintPath>
-    </Reference>
-    <Reference Include="System.Windows.Forms" />
-    <Reference Include="System.Xml" />
-    <Reference Include="System.Xml.Linq" />
-  </ItemGroup>
-  <ItemGroup>
-    <None Include="packages.config" />
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj]: Let's delete the file as well.

File: src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj (hunk 1 above)
Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj]: Will fix.

File: src/Microsoft.DotNet.Build.Tasks/project.json (hunk 2 above)
Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks/project.json]: Good catch! I forgot to remove package.config contents, I just pushed that change as well. Basically we are using project.json now instead for that.

File: src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj (hunk 3 above)
Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj]: I see, I just pushed a change that deletes its contents, but I can just go ahead and delete the file all together as well.

File: src/Microsoft.DotNet.Build.Tasks/project.json [hunk 4]
```diff
@@ -1,8 +1,27 @@
...
   },
   "frameworks": {
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks/project.json]: We may want to have a Net46 target as well, but that won't work well with frameworks + RIDs because it results in a cross-product of all frameworks x all RIDs.  Once we switch to using nuget.exe/dotnet.exe to do the restore in buildtools then this should be built as a v3 portable library with "framework": { "dotnet5.4" } and "supports" clauses for each TFM:RID tuple we want to target.

File: src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj [hunk 5]
```diff
@@ -4,18 +4,13 @@
...
+    <ProjectGuid>{9AB674A0-AEF7-4539-8049-0406D2C7BB32}</ProjectGuid>
     <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>Microsoft.DotNet.Build.Tasks</RootNamespace>
     <AssemblyName>Microsoft.DotNet.Build.Tasks</AssemblyName>
-    <TargetFrameworkIdentifier>.NETFramework</TargetFrameworkIdentifier>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <CopyNuGetImplementations>true</CopyNuGetImplementations>
+    <CLSCompliant>false</CLSCompliant>
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj]: Nit: Is this required?

Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj]: Yes, there is some code in PrereleaseResolveNugetPackageAssets task that would fail otherwise.

Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj]: I meant the CLSCompliant property.  It wasn't there before and you didn't change any source code to use non-CLSCompliant types.

**Comment by ericstj** [comment]
LGTM, thanks for following up!  Sorry I missed the first PR.

File: src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj (hunk 5 above)
Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj]: That's what I meant. I tried removing the property and got some errors saying that some types in that task are not CLSCompliant

**Comment by joperezr** [comment]
Thanks for reviewing and again, sorry for the trouble!
